packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationEventsStreamHandler.kt

```diff
@@ -0,0 +1,108 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.amplify.amplify_push_notifications
+
+import io.flutter.plugin.common.BinaryMessenger
+import io.flutter.plugin.common.EventChannel
+
+private const val channelNamePrefix = "com.amazonaws.amplify/push_notification"
+
+enum class NativeEvent {
+    TOKEN_RECEIVED,
+    NOTIFICATION_OPENED,
+    LAUNCH_NOTIFICATION_OPENED,
+    FOREGROUND_MESSAGE_RECEIVED,
+    BACKGROUND_MESSAGE_RECEIVED;
+
+    val eventName: String
+        get() = when (this) {
+            TOKEN_RECEIVED -> "TOKEN_RECEIVED"
+            NOTIFICATION_OPENED -> "NOTIFICATION_OPENED"
+            LAUNCH_NOTIFICATION_OPENED -> "LAUNCH_NOTIFICATION_OPENED"
+            FOREGROUND_MESSAGE_RECEIVED -> "FOREGROUND_MESSAGE_RECEIVED"
+            BACKGROUND_MESSAGE_RECEIVED -> "BACKGROUND_MESSAGE_RECEIVED"
+        }
+
+    val eventChannelName: String
+        get() = "$channelNamePrefix/event/$eventName"
+}
+
+val eventChannels = listOf(
+    NativeEvent.TOKEN_RECEIVED.eventChannelName,
+    NativeEvent.NOTIFICATION_OPENED.eventChannelName,
+    NativeEvent.LAUNCH_NOTIFICATION_OPENED.eventChannelName,
+    NativeEvent.FOREGROUND_MESSAGE_RECEIVED.eventChannelName,
+    NativeEvent.BACKGROUND_MESSAGE_RECEIVED.eventChannelName
+)
+
+data class PushNotificationsEvent(
+    val event: NativeEvent,
+    val payload: Map<String, Any?>
+) {
+    fun toMap(): Map<String, Any> {
+        return mapOf(
+            "eventType" to event.eventName,
+            "payload" to payload
+        )
+    }
+}
+
+class PushNotificationEventsStreamHandler {
+    companion object : EventChannel.StreamHandler {
+        private val TAG = "PushNotificationEventsStreamHandler"
+
+        fun initialize(binaryMessenger: BinaryMessenger) {
+            eventChannels.forEach { eventChannelName ->
```

Comment:
This is fine for now - but I think one stream handler per event channel still makes more sense